clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -574,6 +697,42 @@ private void assignOwnedPartitions() {
             }
         }
 
+        // Round-Robin filling within racks for remaining members up to the expected numbers of maxQuota,
+        // otherwise, to minQuota
+        private void assignRackAwareRoundRobin(List<TopicPartition> unassignedPartitions) {
+            int nextUnfilledConsumerIndex = 0;
+            Iterator<TopicPartition> unassignedIter = unassignedPartitions.iterator();
+            while (!rackInfo.consumerRacks.isEmpty() && unassignedIter.hasNext()) {
```

Comment:
nit: Do we ever mutate `ackInfo.consumerRacks`? If not, would it make sense to put this check as first statement in the method and return if is it empty?

Reply:
We don't mutate consumerRacks, updated to return early if empty.